src/main/java/MyArrayList.java

```diff
@@ -0,0 +1,335 @@
+    size--;
+    T toRemove = tempTable[index];
+    tempTable = (T[]) createTable;
+    return toRemove;
+  }
+
+  public boolean removeAll(final Collection<?> c) {
+    Objects.requireNonNull(c);
+    boolean result = false;
+    for (Object o : c) {
+      while (this.contains(o)) {
+        this.remove(o);
+        result = true;
+      }
+    }
+    return result;
+  }
+
+  public boolean containsAll(final Collection<?> c) {
+    Objects.requireNonNull(c);
+    int index = 0;
+    List<Object> list = (List<Object>) c;
+    for (Object o : list) {
+      if (this.contains(o)) {
+        index++;
+      }
+    }
+
+    return index >= c.size();
+  }
+
+  public boolean addAll(final Collection<? extends T> c) {
+    return addAll(size, c);
+  }
+
+  public boolean addAll(final int index, final Collection<? extends T> c) {
+    Objects.requireNonNull(c);
+    if (index < 0 || index > size) {
+      return false;
+    }
+    int i = index;
+    for (Object o : c) {
+      this.add(i, (T) o);
+      i++;
+    }
+    return true;
+  }
+
+  public boolean retainAll(final Collection<?> c) {
+    Objects.requireNonNull(c);
+    boolean result = false;
+    for (Object elTemp : tempTable) {
+      if (c.contains(elTemp)) {
+        continue;
+      } else {
+        this.remove(elTemp);
+        result = true;
+      }
+    }
+    return result;
+  }
+
+  public void replaceAll(final UnaryOperator<T> operator) {
+    for (int i = 0; i < size; i++) {
+      tempTable[i] = operator.apply(tempTable[i]);
+    }
+  }
+
+  public void sort(final Comparator<? super T> c) {
+    Arrays.sort(tempTable, c);
+  }
+
+  public void clear() {
+    tempTable = (T[]) new Object[0];
+    size = 0;
+  }
+
+  public T get(final int index) {
+    return tempTable[index];
+  }
+
+  public T set(final int index, final T element) {
+    tempTable[index] = element;
+    return element;
+  }
+
+  public int indexOf(final Object o) {
+    if (o == null) {
+      return -1;
+    }
+    for (int i = 0; i < size; i++) {
+      if (tempTable[i].equals(o)) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  public int lastIndexOf(final Object o) {
+    if (o == null) {
+      return -1;
+    }
+    for (int i = size - 1; i >= 0; i--) {
+      if (tempTable[i].equals(o)) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  public List<T> subList(final int fromIndex, final int toIndex) {
+    if (fromIndex > toIndex) {
+      throw new IllegalArgumentException("fromIndex > toIndex");
+    }
+    if (fromIndex < 0) {
+      throw new IndexOutOfBoundsException("fromIndex = " + fromIndex);
+    }
+    if (toIndex > size) {
+      throw new IndexOutOfBoundsException("toIndex = " + toIndex);
+    }
+    if (fromIndex == toIndex) {
+      return null;
+    }
+
+    T[] table = (T[]) new Object[toIndex - fromIndex];
+
+    for (int i = fromIndex; i < toIndex; i++) {
+      table[i - fromIndex] = tempTable[i];
+    }
+    return new MyArrayList<T>(Arrays.asList(table));
```

Comment:
it's not valid implementation. - please review javadoc. - it says. that any. change in sublist should change. original list and my. making copy. you will. not. achieve it.